README
This should become the BT Web application for checking the items we have in stock.

All the code you will find in here is based on the nice book I bought some time ago
here in Salvador, Brasil. The book is from www.editoraerica.com.br , code 2373,
ISBN: 978-85-365-0237-3 , name of the book is:

 Crie um Framework para Sistemas Web com PHP5 e AJAX.

Translated is Create a framework for web systems with PHP5 and AJAX.
Longer I go in the book more I will update here.

Sasa Ostrouska - 28-05-2012 


btwebstock/ChangeLog
2012-07-08 Sasa Ostrouska <casaxa@gmail.com>

	* Added class_pagination.inc into classes/
	* Added test_pagination.php in tests dir.
	* Added pagination functionality to class Base and class DataBase.
	* Included class_pagination.inc in test_psql_class_base.php.

2012-06-24 Sasa Ostrouska <casaxa@gmail.com>

	* Renamed all calls from database to DataBase.
	* Added test_psql_include_registers.php in tests dir.

2012-06-22 Sasa Ostrouska <casaxa@gmail.com>

	* Fixed a typo in class_base.inc 
	* Added some missing code in __call methond to be able to handle well
	set* methods.

2012-06-17 Sasa Ostrouska <casaxa@gmail.com>

	* Renamed class base into Base in class_base.inc
	* Added methods deleteField(), clearFields() and getField() to class
	Base.
	* Added methods filterFields() and createSELECT() to class Base.
	* Added methods list(), fillField() and nextField() to class Base.
	* Updated method __call to be able to filter get and set function
	names.
	* Renamed method datetime() to date_time() to avoid conflicts.
	* Renamed method list() to listStuff().
	* Added in tests a test_psql_class_base.php to test the new Base
	class.
	* Fixed a typo in class field with a missing _ in _fieldref.

2012-06-12 Sasa Ostrouska <casaxa@gmail.com>

	* Added class date and datetime to the class_base.inc.
	* Added methods convertDateToDB(), convertDateTimeToDB(),
	convertDateToHTML() and convertDateTimeToHTML() to the DataBase class 
	in class_database.inc.
	* Added method __call in class field. 
	* Started writing class base in class_base.inc

2012-06-10 Sasa Ostrouska <casaxa@gmail.com>
	
	* Started writing class_base.inc
	* Added class field and string.
	* Added method escapeString in database class.
	* Added classes integer, float and money into class_base.inc

2012-06-07 Sasa Ostrouska <casaxa@gmail.com>

	* Added class tag and class html in class_html.inc
	* Added in tests test_psql_select_using_htmlclass.php, which creates
	dynamically the HTML code for the showed page.
	* Added in tests test_mysql_select_using_htmlclass.php, which creates
	dynamically the HTML code for the showed page.
	* Added html optimization code to the class_html.inc, this will output
	a way more readable code.

2012-06-05 Sasa Ostrouska <casaxa@gmail.com>

	* Added test_psql_transaction_test1.php 
	* Added test_psql_transaction_test2.php 
	* Renamed index_mysql_transaction_test{1,2}.php into
	test_mysql_transaction_test{1,2}.php
	* Fixd test_mysql_transaction_test1.php
	* Fixd test_mysql_transaction_test2.php
	* Renmed some function with a clearer name, an fixed some stuff in the
	DataBase class.
	* Added the test for error in the SQL querry.
	* Started writing the class_html.inc in classes/

2012-06-04 Sasa Ostrouska <casaxa@gmail.com>

	* Added skeletions dir, and an empty php skeletion file.
	* Added GNU GPL licence header in class_database.php

2012-05-31 Sasa Ostrouska <casaxa@gmail.com>

	* Created a ChangeLog file.


btwebstock/htdocs/tests/test_psql.php
<?php

/*
 * Example of database use
 *
 */
$debug = 1;

$currentdir = getcwd();
$includesdir = $currentdir."/../classes/";

if ($debug){
	echo $includesdir."<br>";
}

include_once($includesdir."class_database.inc");

$_db = new pgsql();
$_db->SetServer('localhost');
$_db->SetDBase('db_btwebstock');
$_db->SetUser('postgres');
$_db->SetPassword('postgres');
$_db->SetPort(5432);
/* if (var_dump($_db->Connect())) {
	echo "You have fucked up something. Go and debug.";
} */
$_db->Connect();

$_sql = "create table tab_test (code int4 default 0, description varchar(20), value float, primary key(code))";
if($_db->executeSQL($_sql) !== false) {
	echo "Table sucessfully created !<br>";
}

$_sql = "insert into tab_test (code, description, value) values (1, 'test', 1.5)";
if($_db->executeSQL($_sql) !== false) {
	echo "1st register inserted.<br>";
}

$_sql = "insert into tab_test (code, description, value) values (2, 'test', 3.5)";
if($_db->executeSQL($_sql) !== false) {
	echo "2nd register inserted.<br>";
}

$_sql = "insert into tab_test (code, description, value) values (3, 'test', 7.15)";
if($_db->executeSQL($_sql) !== false) {
	echo "3rd register inserted.<br>";
}


?>



btwebstock/htdocs/tests/test_mysql.php
<?php

/*
 * Example of database use
 *
 */

$debug = 1;

$currentdir = getcwd();
$includesdir = $currentdir."/../classes/";

if ($debug){
	echo $includesdir."<br>";
}

include_once($includesdir."class_database.inc");

$_db = new mysql();
$_db->SetServer('localhost');
$_db->SetDBase('db_btwebstock_test');
$_db->SetUser('root');
$_db->SetPassword('mysqlrootpa55word');
$_db->SetPort(3306);
$_db->Connect();


$_sql = "create table tab_test(
	code int4 default 0, 
	description varchar(20), 
	value float, 
	primary key(code))";
if($_db->executeSQL($_sql)!== false) {
	echo "Table sucessfully created !<br>";
}

$_sql = "insert into tab_test(code, description, value) values (1, 'testdesc', 1.55)";
if($_db->executeSQL($_sql) !== false) {
	echo "1st register inserted.<br>";
}

$_sql = "insert into tab_test(code, description, value) values (2, 'testdesc1', 11.65)";
if($_db->executeSQL($_sql) !== false) {
	echo "2nd register inserted.<br>";
}

$_sql = "insert into tab_test(code, description, value) values (3, 'testdesc2', 81.45)";
if($_db->executeSQL($_sql) !== false) {
	echo "3rd register inserted.<br>";
}

?>


btwebstock/htdocs/tests/index_mysql.php
<?php

/*
 * Example of database use
 *
 */

$debug = 1;

$currentdir = getcwd();
$includesdir = $currentdir."/../classes/";

if ($debug){
	echo $includesdir."<br>";
}

include_once($includesdir."class_database.inc");

$_db = new mysql();
$_db->SetServer('localhost');
$_db->SetDBase('db_btwebstock_test');
$_db->SetUser('root');
$_db->SetPassword('mysqlrootpa55word');
$_db->SetPort(3306);
$_db->Connect();


$_sql = "create table table_test(
	code int4 default 0, 
	description varchar(20), 
	value float, 
	primary key(code))";
if($_db->executeSQL($_sql)!== false) {
	echo "Table sucessfully created !<br>";
}

$_sql = "insert into table_test(code, description, value) values (1, 'testdesc', 1.55)";
if($_db->executeSQL($_sql) !== false) {
	echo "1st register inserted.<br>";
}

$_sql = "insert into table_test(code, description, value) values (2, 'testdesc1', 11.65)";
if($_db->executeSQL($_sql) !== false) {
	echo "2nd register inserted.<br>";
}

$_sql = "insert into table_test(code, description, value) values (3, 'testdesc2', 81.45)";
if($_db->executeSQL($_sql) !== false) {
	echo "3rd register inserted.<br>";
}

?>


btwebstock/htdocs/index_mysql.php
<?php

/*
 * Example of database use
 *
 */

$debug = 0;

$currentdir = getcwd();
$includesdir = $currentdir."/classes/";

if ($debug){
	echo $includesdir."<br>";
}

include_once($includesdir."class_database.inc");

$_db = new mysql();
$_db->SetServer('localhost');
$_db->SetDBase('db_btwebstock');
$_db->SetUser('root');
$_db->SetPassword('mysqlrootpa55word');
$_db->SetPort(3306);
$_db->Connect();

$_sql = "create table tab_test(code int4 default 0, description varchar(20), value float, primary key(code))";
if($_db->executeSQL($_sql)!== false) {
	echo "Table sucessfully created !<br>";
}

$_sql = "insert into tab_test (description) values ('test')";
if($_db->executeSQL($_sql) !== false) {
	echo "1st register inserted.<br>";
}

$_sql = "insert into tab_test (description) values ('test')";
if($_db->executeSQL($_sql) !== false) {
	echo "2nd register inserted.<br>";
}
$_sql = "insert into tab_test (description) values ('test')";
if($_db->executeSQL($_sql) !== false) {
	echo "3rd register inserted.<br>";
}

?>


btwebstock/htdocs/index_psql_select.php
<?php

/*
 * Example of database use
 *
 */
$debug = 0;

$currentdir = getcwd();
$includesdir = $currentdir."/classes/";

if ($debug){
	echo $includesdir."<br>";
}

include_once($includesdir."class_database.inc");

$_db = new pgsql();
$_db->SetServer('localhost');
$_db->SetDBase('db_btwebstock');
$_db->SetUser('postgres');
$_db->SetPassword('postgres');
$_db->SetPort(5432);
//var_dump($_db->Connect());
//	echo "You have fucked up something. Go and debug.";
$_db->Connect() or die($_db->getLastError());

$_sql = "SELECT * FROM tab_test";
$_db->executeSQL($_sql);

echo "Number of registers returned by SELECT:{$_db->getNumRows()}<br>";
echo "<table border=1 cellpadding=5 width=400>
	<tr><th>Code</th><th>Description</th><th>Value</th></tr>";
while($_d = $_db->nextPos()) {
	echo "<tr><td>{$_d['code']}<td>
		<td>{$_d['description']}</td>
		<td align='right'>{$_d['value']}</td></tr>";
}
echo "</table>";

?>


btwebstock/htdocs/index_mysql_select.php
<?php

/*
 * Example of database use
 *
 */
$debug = 1;

$currentdir = getcwd();
$includesdir = $currentdir."/classes/";

if ($debug){
	echo $includesdir."<br>";
}

include_once($includesdir."class_database.inc");

$_db = new mysql();
$_db->SetServer('localhost');
$_db->SetDBase('db_btwebstock_test');
$_db->SetUser('root');
$_db->SetPassword('mysqlrootpa55word');
$_db->SetPort(3306);
$_db->Connect();

$_sql = "SELECT * FROM table_test";
$_db->executeSQL($_sql);

echo "Number of registers returned by SELECT:{$_db->getNumRows()}<br>";
echo "<table border=1 cellpadding=5 width=400>
	<tr>
		<th>Code</th>
		<th>Description</th>
		<th>Value</th>
	</tr>";
while($_d = $_db->nextPos()) {
	echo "	<tr>
			<td>{$_d['code']}</td>
			<td>{$_d['description']}</td>
			<td align='right'>{$_d['value']}</td>
		</tr>";
}
echo "</table>";

?>


btwebstock/htdocs/index_psql.php
<?php

/*
 * Example of database use
 *
 */
$debug = 1;

$currentdir = getcwd();
$includesdir = $currentdir."/classes/";

if ($debug){
	echo $includesdir."<br>";
}

include_once($includesdir."class_database.inc");

$_db = new pgsql();
$_db->SetServer('localhost');
$_db->SetDBase('db_btwebstock');
$_db->SetUser('postgres');
$_db->SetPassword('postgres');
$_db->SetPort(5432);
/* if (var_dump($_db->Connect())) {
	echo "You have fucked up something. Go and debug.";
} */
$_db->Connect();

$_sql = "create table tab_test (code int4 default 0, description varchar(20), value float, primary key(code))";
if($_db->executeSQL($_sql) !== false) {
	echo "Table sucessfully created !<br>";
}

$_sql = "insert into tab_test (code, description, value) values (1, 'test', 1.5)";
if($_db->executeSQL($_sql) !== false) {
	echo "1st register inserted.<br>";
}

$_sql = "insert into tab_test (code, description, value) values (2, 'test', 3.5)";
if($_db->executeSQL($_sql) !== false) {
	echo "2nd register inserted.<br>";
}

$_sql = "insert into tab_test (code, description, value) values (3, 'test', 7.15)";
if($_db->executeSQL($_sql) !== false) {
	echo "3rd register inserted.<br>";
}


?>

